Nil pointer dereference in new Jetstream API when getting consumer

### What version were you using?

server: `2.9.15`
client: `1.28.0`


### What environment was the server running in?

Server running on x86 from `2.9.15-alpine` container

### Is this defect reproducible?

Has occurred multiple times during testing. Working towards reproducing, havent gotten a minimal example yet.

### Given the capability you are leveraging, describe your expectation?

There are multiple pods running the same code, and each pod will attempt to get a certain consumer by name and if the consumer doesnt exist, will attempt to create the consumer.

This bug tends to occur on pod startup, where it have a list of consumers to try and get/create. This bug occured typically when working with 500- 10,000 consumers. 

The consumers are on a work queue, so the `CreateOrUpdate` command does not work, as it returns the error about filters needing to be unique on work queue. The consumers are ephemeral.  

The expectation is that there are no crashes due to nil pointer dereferences.

### Given the expectation, what is the defect you are observing?

```
panic: runtime error: invalid memory address or nil pointer dereference                                                           │
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0xbfcb19]                                                           │
                                                                                                                                  │
goroutine 6151 [running]:                                                                                                         │
github.com/nats-io/nats.go/jetstream.getConsumer({0x1dbca10?, 0xc00007cc80?}, 0xc001635140, {0x1953ca0, 0x2}, {0xc002ba48a0, 0xa})│
        /home/users/eslack/go/pkg/mod/github.com/nats-io/nats.go@v1.28.0/jetstream/consumer.go:178 +0x499                         │
github.com/nats-io/nats.go/jetstream.(*jetStream).Consumer(0xc00007f620?, {0x1dbca10, 0xc00007cc80}, {0x1953ca0, 0x2}, {0xc002ba48│
        /home/users/eslack/go/pkg/mod/github.com/nats-io/nats.go@v1.28.0/jetstream/jetstream.go:512 +0x8d                         
```

--- a/jetstream/consumer.go
+++ b/jetstream/consumer.go
@@ -77,6 +77,9 @@ func (p *pullConsumer) Info(ctx context.Context) (*ConsumerInfo, error) {
 		}
 		return nil, resp.Error
 	}
+	if resp.Error == nil && resp.ConsumerInfo == nil {
+		return nil, ErrConsumerNotFound
+	}
 
 	p.info = resp.ConsumerInfo
 	return resp.ConsumerInfo, nil
@@ -187,6 +190,9 @@ func getConsumer(ctx context.Context, js *jetStream, stream, name string) (Consu
 			return nil, ErrConsumerNotFound
 		}
 		return nil, resp.Error
+	}
+	if resp.Error == nil && resp.ConsumerInfo == nil {
+		return nil, ErrConsumerNotFound
 	}
 
 	cons := &pullConsumer{

--- a/jetstream/ordered.go
+++ b/jetstream/ordered.go
@@ -519,6 +519,9 @@ func (c *orderedConsumer) Info(ctx context.Context) (*ConsumerInfo, error) {
 		}
 		return nil, resp.Error
 	}
+	if resp.Error == nil && resp.ConsumerInfo == nil {
+		return nil, ErrConsumerNotFound
+	}
 
 	c.currentConsumer.info = resp.ConsumerInfo
 	return resp.ConsumerInfo, nil

--- a/js.go
+++ b/js.go
@@ -3223,6 +3223,9 @@ func (js *js) getConsumerInfoContext(ctx context.Context, stream, consumer strin
 		}
 		return nil, info.Error
 	}
+	if info.Error == nil && info.ConsumerInfo == nil {
+		return nil, ErrConsumerNotFound
+	}
 	return info.ConsumerInfo, nil
 }
 

